hook.go

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/golang/glog"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// Hook represents an OCI hook, including its required parameters.
+type Hook struct {
+	Path    string
+	Args    []string
+	Env     []string
+	Timeout int
+}
+
+// Hooks gathers all existing OCI hooks list.
+type Hooks struct {
+	PreStartHooks  []Hook
+	PostStartHooks []Hook
+	PostStopHooks  []Hook
+}
+
+func buildHookState(processID int) specs.State {
+	return specs.State{
+		Pid: processID,
+	}
+}
+
+func (h *Hook) runHook() error {
+	state := buildHookState(os.Getpid())
+	stateJSON, err := json.Marshal(state)
+	if err != nil {
+		return err
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := &exec.Cmd{
```

Comment:
@sboeuf Is the PID in this state file correct. You need to pass the PID of a process that is in the network namespace you created. I think you are in the parent ns here. I may be mistaken though. Is this code working?